src/Broadway/CommandHandling/ClosureCommandHandler.php

```diff
@@ -31,13 +32,22 @@ public function add(\Closure $handler): void
         $reflection = new \ReflectionFunction($handler);
         $reflectionParams = $reflection->getParameters();
 
-        if (!isset($reflectionParams[0]) || !$reflectionParams[0]->getClass()) {
-            throw new ClosureParameterNotAnObjectException();
+        $reflectionType = $reflectionParams[0]->getType();
```

Comment:
Doesn't this fail when there are no parameters? This is what we need a test for.

Reply:
Ok, thanks for the clarification, I will try to add tests 👍

Reply:
I Have tried something but it seems that if there are no params it's already covered by `it_throws_when_adding_a_closure_without_an_object_argument` because if there aren't parameters inside `$reflection->getParameters()` there is always something like:

```
array(1) {
  [0]=>
  object(ReflectionParameter)#853 (1) {
    ["name"]=>
    string(7) "command"
  }
}
```

So that test passes because after it doesn't find the correct type and var `$name ` still null. 
So it seems that when there are no parameters it's already covered or I am missing something? 
We can add an if condition to be more defensive that checks if `$reflectionParams` is not empty for example but at the moment it's always filled but the command name for example. 
What do you think?

Reply:
Well, there is one parameter in that test: `$params`.

What happens if you add a test like this?

```
/**
     * @test
     */
    public function it_throws_when_adding_a_closure_without_an_object_argument()
    {
        $commandHandler = new ClosureCommandHandler();
        $this->expectException(ClosureParameterNotAnObjectException::class);

        $commandHandler->add(function () { });
    }
```

Reply:
You are right!
I have added that test and an if condition to check this behavior!